File: src/main/java/pe/nanamochi/banchus/entities/commons/MatchStatus.java
package pe.nanamochi.banchus.entities.commons;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum MatchStatus {
  WAITING(0),
  PLAYING(1);

  private final int value;

  public static MatchStatus fromValue(int value) {
    for (MatchStatus matchStatus : values()) {
      if (matchStatus.value == value) {
        return matchStatus;
      }
    }
    throw new IllegalArgumentException("Unknown status value: " + value);
  }
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/domain/enums/MatchStatus.kt
package pe.nanamochi.banchus.domain.enums

enum class MatchStatus(val value: Int) {
    WAITING(0),
    PLAYING(1);

    companion object {
        fun fromValue(value: Int): MatchStatus = entries.find { it.value == value } ?: WAITING
    }
}
